ocaml/utils/language_extension.ml

```diff
@@ -77,47 +51,38 @@ let get_level_ops : type a. a t -> (module Extension_level with type t = a) =
   | Module_strengthening -> (module Unit)
   | Layouts -> (module Maturity)
 
+type extn_pair = Pair : 'a t * 'a -> extn_pair
+type exist = Exist.t = Pack : _ t -> exist
+
 (**********************************)
 (* string conversions *)
 
-let to_string : type a. a t -> string = function
-  | Comprehensions -> "comprehensions"
-  | Local -> "local"
-  | Include_functor -> "include_functor"
-  | Polymorphic_parameters -> "polymorphic_parameters"
-  | Immutable_arrays -> "immutable_arrays"
-  | Module_strengthening -> "module_strengthening"
-  | Layouts -> "layouts"
-
 (* converts full extension names, like "layouts_alpha" to a pair of
-   an extension and its setting *)
-let pair_of_string extn_name : extn_pair option =
-  match String.lowercase_ascii extn_name with
-  | "comprehensions" -> Some (Pair (Comprehensions, ()))
-  | "local" -> Some (Pair (Local, ()))
-  | "include_functor" -> Some (Pair (Include_functor, ()))
-  | "polymorphic_parameters" -> Some (Pair (Polymorphic_parameters, ()))
-  | "immutable_arrays" -> Some (Pair (Immutable_arrays, ()))
-  | "module_strengthening" -> Some (Pair (Module_strengthening, ()))
-  | "layouts" -> Some (Pair (Layouts, (Stable : Maturity.t)))
-  | "layouts_beta" -> Some (Pair (Layouts, (Beta : Maturity.t)))
-  | "layouts_alpha" -> Some (Pair (Layouts, (Alpha : Maturity.t)))
-  | _ -> None
+   an extension and its setting. For extensions that don't take an
```

Comment:
Is "setting" the right word here?  I think we mean "maturity"?

Reply:
```suggestion
   an extension and its maturity. For extensions that don't take an
```